src/main/java/seedu/address/logic/commands/CloseTutorialCommand.java

```diff
@@ -0,0 +1,106 @@
+
+    private final Tutorial toClose;
+
+    /**
+     * Constructs a {@code CloseTutorialCommand} with the specified tutorial to close.
+     *
+     * @param tutorial The tutorial to close.
+     */
+    public CloseTutorialCommand(Tutorial tutorial) {
+        requireNonNull(tutorial);
+        toClose = tutorial;
+    }
+
+    /**
+     * Executes the command to close a tutorial. If the tutorial exists, all students will be unenrolled,
+     * and the tutorial will be marked as closed. (Check UnEnrollCommand to see its implementation)
+     *
+     * @param model {@code Model} which the command operates on.
+     * @return CommandResult which indicates the success of the tutorial closure.
+     * @throws CommandException if the tutorial does not exist.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (!model.hasTutorial(toClose)) {
+            throw new CommandException(String.format(MESSAGE_TUTORIAL_NOT_FOUND, toClose.getSubject()));
+        }
+
+        Tutorial tutorialToCloseFromList = model.getTutorial(toClose);
+        removeStudentsFromTutorialParticipation(model, tutorialToCloseFromList);
+        model.closeTutorial(tutorialToCloseFromList);
+
+        return new CommandResult(String.format(
+                MESSAGE_CLOSE_TUTORIAL_SUCCESS,
+                Messages.formatTutorial(tutorialToCloseFromList))
+        );
+    }
+
+    /**
+     * Unenrolls all students from the given tutorial by executing {@code UnEnrollCommand} for each student.
+     *
+     * @param model    The {@code Model} on which the command operates.
+     * @param tutorial The tutorial from which students will be unenrolled.
+     * @throws CommandException if an error occurs during the unenrollment process.
+     */
+    private void removeStudentsFromTutorialParticipation(Model model, Tutorial tutorial) throws CommandException {
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        List<Person> lastShownStudentList = model.getFilteredPersonList();
+        List<Participation> tutorialParticipationCopy = new ArrayList<>(tutorial.getParticipationList());
+        for (Participation eachParticipation : tutorialParticipationCopy) {
+            Person student = eachParticipation.getStudent();
+            Index studentIndex = getIndexOfPerson(lastShownStudentList, student);
+            new UnEnrollCommand(studentIndex, tutorial.getSubject()).execute(model);
+        }
+    }
+
+    /**
+     * Finds the index of the given student in a list of all students.
+     *
+     * @param allPersons The list of all persons in the system.
+     * @param student    The student whose index is to be found.
+     * @return The index of the student.
+     */
+    private Index getIndexOfPerson(List<Person> allPersons, Person student) {
+        int index = allPersons.indexOf(student);
+        if (index == -1) {
```

Comment:
Just curious, why is the exception thrown for index ++ -1 instead of index < 0?

Reply:
because the indexOf method returns the index of the object if the object is present and -1 if the object is not found in the list